docs/root/configuration/http/http_filters/on_demand_updates_filter.rst

```diff
@@ -0,0 +1,16 @@
+.. _config_http_filters_on_demand:
+
+On-demand VHDS Updates
+======================
+
+The on-demand VHDS filter is used to request a :ref:`virtual host <envoy_api_msg_route.VirtualHost>`
+data if it's not already present in the :ref:`Route Configuration <envoy_api_msg_RouteConfiguration>`. The
+contents of the *Host* or *:authority* header is used to create the on-demand request. For an on-demand
+request to be created, :ref:`VHDS <envoy_api_field_RouteConfiguration.vhds>` must be enabled and either *Host*
+or *:authority* header be present.
+
+Configuration
+-------------
+* :ref:`v2 API reference <envoy_api_msg_config.filter.http.on_demand.v2.OnDemand>`
+* The filter name is *envoy.on_demand*. The On Demand VHDS Updates filter is automatically configured at the
```

Comment:
This is where we should switch to providing explicit documentation on configuration. I think there are legitimate reasons why folks might want to run their own filters before VHDS.

Reply:
> I think there are legitimate reasons why folks might want to run their own filters before VHDS.

That's fair. Made the filter configurable